tasks/benchmark.rake

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+
+desc 'Runs a performance benchmark on the SDK'
+task 'benchmark' do
+  sh('bundle exec ruby benchmark/run_benchmark.rb')
+end
+
+desc 'Upload/archive the benchmark report'
+task 'benchmark:archive' do
+  $:.unshift("#{$GEMS_DIR}/aws-sdk-s3/lib")
+  $:.unshift("#{$GEMS_DIR}/aws-sdk-kms/lib")
+  require 'aws-sdk-s3'
+  require 'securerandom'
+
+
+  puts "Archiving benchmark report from GH with: #{ENV['GH_REF']}, #{ENV['GH_EVENT']}"
+  ref = ENV['GH_REF'] || "version-3"
+  event = ENV['GH_EVENT'] || 'push'
+  client = Aws::S3::Client.new
+
+  key = "#{event == 'pull_request' ? 'pr/' + ref : 'release'}/#{Time.now.strftime('%Y-%m-%d')}/benchmark_#{SecureRandom.uuid}.json"
+  puts "Uploading report to: #{key}"
+  client.put_object(bucket: 'aws-sdk-ruby-performance-benchmark-archive', key: key, body: File.read('benchmark_report.json'))
+  puts "Upload complete"
+end
+
+desc 'Upload benchmarking data to cloudwatch'
+task 'benchmark:put-metrics' do
+  $:.unshift("#{$GEMS_DIR}/aws-sdk-cloudwatch/lib")
+  require 'aws-sdk-cloudwatch'
+  require_relative '../benchmark/metrics'
+
+  event = ENV['GH_EVENT'] == 'pull_request' ? 'pr' : 'release'
+  report = JSON.parse(File.read('benchmark_report.json'))
+  target = report['ruby_engine'] + "-" + report['ruby_version'].split('.').first(2).join('.') # TODO: How do we want to capture the environment
```

Comment:
It's the major and minor version.  It drops the third/patch version to prevent metrics churn on minor version updates.

Its fairly robust because the split/first/join method all work on a range of inputs.  Eg, if ruby version is "beta", the target will be "ruby-beta".  If its "1.2.3.4.5.beta", it will be "ruby-1.2".

Reply:
I know with rails they have constants for major/minor. Maybe there's a ruby constant for those too. But this is otherwise fine.

Reply:
I think the TODO can be removed, then, seems like this is the best way.